### PR TITLE
remake with symbols infrastructure

### PR DESCRIPTION
**Pull request by xtalax**
Infrastructure for remaking with symbols where appropriate

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This isn't split appropriately and will heavily invalidate. Instead, make this drop the arg `prob.f.sys` appropriately (checking for existence etc.) in a subfunction, and extend that on a dispatch for `AbstractSystem`

File: src/remake.jl [hunk 1]
```diff
@@ -59,7 +59,13 @@ function remake(prob::ODEProblem; f = missing,
     if p === missing
         p = prob.p
     else
-        p = handle_varmap(p, prob)
+        @show typeof(prob.f.sys)
+        @show typeof(prob.f.sys.ps)
+        if hasproperty(prob.f, :sys) && hasproperty(prob.f.sys, :ps)
+            p = handle_varmap(p, prob.f.sys)
```
Comment (ChrisRackauckas) [inline src/remake.jl:65]: you're missing the basic dispatch here

File: src/remake.jl [hunk 2]
```diff
@@ -58,6 +58,15 @@ function remake(prob::ODEProblem; f = missing,
 
     if p === missing
         p = prob.p
+    else
+        @show typeof(prob.f.sys)
+        @show hasproperty(prob.f, :sys)
+        @show hasproperty(prob.f.sys, :ps)
+        if hasproperty(prob.f, :sys) && hasproperty(prob.f.sys, :ps)
```
Comment (YingboMa) [inline src/remake.jl:65]: We should handle `u0` as well.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Format

**Review by YingboMa — CHANGES_REQUESTED** [review]
`prob.f.sys.xxx` is wrong. getproperty of a system only means to get the variable in the system. You should call `getfield` directly.

File: src/remake.jl [hunk 3]
```diff
@@ -50,6 +50,14 @@ function remake(prob::ODEProblem; f = missing,
                 _kwargs...)
     if u0 === missing
         u0 = prob.u0
+    else
+        if u0 isa Dict || eltype(u0) isa Pair
+            if hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :states)
+                u0 = handle_varmap(u0, prob.f.sys, field = :states, defaults = prob.f.sys.defaults)
```
Comment (YingboMa) [inline src/remake.jl:56]: Shouldn't the default be constructed from the current states and parameters that the problem holds?

File: src/remake.jl [hunk 4]
```diff
@@ -58,6 +66,14 @@ function remake(prob::ODEProblem; f = missing,
 
     if p === missing
         p = prob.p
+    else
+        if p isa Dict || eltype(p) isa Pair
+            if hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :ps)
+                p = handle_varmap(p, prob.f.sys, field = :ps, defaults = prob.f.sys.ps .=> prob.p)
```
Comment (YingboMa) [inline src/remake.jl:72]: This parameters could depend on the initial condition, so you should include `prob.u0` map as well.

File: src/remake.jl [hunk 5]
```diff
@@ -50,6 +50,15 @@ function remake(prob::ODEProblem; f = missing,
                 _kwargs...)
     if u0 === missing
         u0 = prob.u0
+    else
+        if u0 isa Dict || eltype(u0) isa Pair
+            if hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :states)
+                u0 = handle_varmap(u0, prob.f.sys, field = :states,
+                                   defaults = getfield(prob.f.sys, :states) .=> prob.u0)
```
Comment (ChrisRackauckas) [inline src/remake.jl:57]: this should also have p

**Pull request merged**